File: iosApp/iosApp/Models/ExerciseUnit.swift
import Foundation

struct ExerciseUnit: Identifiable {
    let id: Int
    let name: String
    let exercises: [String]
    var completions: Int
    var isUnlocked: Bool

    var medal: String? {
        switch completions {
        case 5...: return "🥇"
        case 4: return "🥈"
        case 3: return "🥉"
        default: return nil
        }
    }

    static let allUnits: [ExerciseUnit] = [
        ExerciseUnit(id: 1, name: "Unit 1", exercises: ["MIE", "4-0-4", "FloatBall"], completions: 0, isUnlocked: true),
        ExerciseUnit(id: 2, name: "Unit 2", exercises: ["4-1-5", "Candle", "Windmill"], completions: 0, isUnlocked: false),
        ExerciseUnit(id: 3, name: "Unit 3", exercises: ["4-2-5", "Windmill", "Dandelion"], completions: 0, isUnlocked: false),
        ExerciseUnit(id: 4, name: "Unit 4", exercises: ["4-2-5", "Candle"], completions: 0, isUnlocked: false),
        ExerciseUnit(id: 5, name: "Unit 5", exercises: ["4-7-11", "Windmill"], completions: 0, isUnlocked: false),
        ExerciseUnit(id: 6, name: "Unit 6", exercises: ["4-3-5", "Windmill", "Candle"], completions: 0, isUnlocked: false),
        ExerciseUnit(id: 7, name: "Unit 7", exercises: ["4-3-5", "Windmill", "Candle"], completions: 0, isUnlocked: false),
        ExerciseUnit(id: 8, name: "Unit 8", exercises: ["4-7-11", "Candle"], completions: 0, isUnlocked: false),
        ExerciseUnit(id: 9, name: "Unit 9", exercises: ["4-7-11", "Windmill", "Dandelion"], completions: 0, isUnlocked: false),
    ]
}

File: androidApp/src/androidMain/kotlin/com/cloudhaus/sensorapp/model/ExerciseUnit.kt
package com.cloudhaus.sensorapp.model

data class ExerciseUnit(
    val id: Int,
    val name: String,
    val exercises: List<String>,
    val completions: Int = 0,
    val isUnlocked: Boolean = false,
) {
    val medal: String?
        get() = when {
            completions >= 5 -> "\uD83E\uDD47" // gold
            completions == 4 -> "\uD83E\uDD48" // silver
            completions == 3 -> "\uD83E\uDD49" // bronze
            else -> null
        }

    companion object {
        val allUnits = listOf(
            ExerciseUnit(1, "Unit 1", listOf("MIE", "4-0-4", "FloatBall"), isUnlocked = true),
            ExerciseUnit(2, "Unit 2", listOf("4-1-5", "Candle", "Windmill")),
            ExerciseUnit(3, "Unit 3", listOf("4-2-5", "Windmill", "Dandelion")),
            ExerciseUnit(4, "Unit 4", listOf("4-2-5", "Candle")),
            ExerciseUnit(5, "Unit 5", listOf("4-7-11", "Windmill")),
            ExerciseUnit(6, "Unit 6", listOf("4-3-5", "Windmill", "Candle")),
            ExerciseUnit(7, "Unit 7", listOf("4-3-5", "Windmill", "Candle")),
            ExerciseUnit(8, "Unit 8", listOf("4-7-11", "Candle")),
            ExerciseUnit(9, "Unit 9", listOf("4-7-11", "Windmill", "Dandelion")),
        )
    }
}
